src/assets/wise5/components/aiChat/ai-chat-student/ai-chat-student.component.ts

```diff
@@ -51,12 +58,96 @@ export class AiChatStudentComponent extends ComponentStudent {
 
   ngOnInit(): void {
     super.ngOnInit();
+    if (this.component.isComputerAvatarEnabled()) {
+      this.initializeComputerAvatar();
+    } else {
+      this.computerAvatar = this.computerAvatarService.getDefaultAvatar();
+    }
     if (this.componentState != null) {
       this.messages = this.componentState.studentData.messages;
     }
     this.initializeMessages();
   }
 
+  initializeComputerAvatar(): void {
+    this.tryToRepopulateComputerAvatar();
+    if (this.hasStudentPreviouslyChosenComputerAvatar()) {
+      this.hideComputerAvatarSelector();
+    } else if (
+      this.component.isOnlyOneComputerAvatarAvailable() &&
+      !this.component.isComputerAvatarPromptAvailable()
+    ) {
+      this.hideComputerAvatarSelector();
+      this.selectComputerAvatar(this.getTheOnlyComputerAvatarAvailable());
+    } else {
+      this.showComputerAvatarSelector();
+    }
+  }
+
+  private tryToRepopulateComputerAvatar(): void {
+    if (this.includesComputerAvatar(this.componentState)) {
+      this.repopulateComputerAvatarFromComponentState(this.componentState);
+    } else if (
+      this.component.isUseGlobalComputerAvatar() &&
+      this.isGlobalComputerAvatarAvailable()
+    ) {
+      this.repopulateGlobalComputerAvatar();
+    }
+  }
+
+  private includesComputerAvatar(componentState: any): boolean {
+    return componentState?.studentData?.computerAvatarId != null;
+  }
+
+  private isGlobalComputerAvatarAvailable(): boolean {
+    return this.studentStatusService.getComputerAvatarId() != null;
+  }
+
+  private repopulateComputerAvatarFromComponentState(componentState: any): void {
+    this.computerAvatar = this.computerAvatarService.getAvatar(
+      componentState?.studentData?.computerAvatarId
+    );
+  }
+
+  private repopulateGlobalComputerAvatar(): void {
+    const computerAvatarId = this.studentStatusService.getComputerAvatarId();
+    if (computerAvatarId != null) {
+      this.selectComputerAvatar(this.computerAvatarService.getAvatar(computerAvatarId));
+    }
+  }
+
+  private hasStudentPreviouslyChosenComputerAvatar(): boolean {
+    return this.computerAvatar != null;
+  }
+
+  private getTheOnlyComputerAvatarAvailable(): ComputerAvatar {
+    return this.computerAvatarService.getAvatar(
+      this.component.content.computerAvatarSettings.ids[0]
+    );
+  }
+
+  private showComputerAvatarSelector(): void {
+    this.computerAvatarSelectorVisible = true;
+  }
+
+  private hideComputerAvatarSelector(): void {
+    this.computerAvatarSelectorVisible = false;
+  }
+
+  selectComputerAvatar(computerAvatar: ComputerAvatar): void {
```

Comment:
Add access modifier?
```suggestion
  protected selectComputerAvatar(computerAvatar: ComputerAvatar): void {
```